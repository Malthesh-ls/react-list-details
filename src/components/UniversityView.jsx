import { useState } from 'react';
import { config } from '../config/config';
import { columns } from '../config/constants';
import Table from '../components/table/Table';
import SearchBar from './SearchBar';

const UniversityView = ({
  data,
  setData,
  error,
}) => {
  const [searchData, setSearchData] = useState(data);

  const handleDelete = (universityId) => {
    setSearchData(searchData.filter(item => item.uid !== universityId))
    setData(data.filter(item => item.uid !== universityId))
  }

  return (
    <>
      {!error
        ?
        <>
          {data.length > 0 ?
            <>
              <SearchBar data={data} setSearchData={setSearchData} />
              <Table
                columns={columns}
                data={searchData}
                handleDelete={handleDelete}
              />
            </>
            : <div>No data found</div>
          }
        </>
        : <div className='errMsg alignCenter'>{config.errMsg}</div>
      }
    </>
  )
}

export default UniversityView