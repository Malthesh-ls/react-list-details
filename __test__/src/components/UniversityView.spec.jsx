import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import UniversityView from '../../../src/components/UniversityView';

describe('UniversityView Tests', () => {
  const props = {
    data: [],
    setData: jest.fn(),
    error: false,
  }
  it('should match snapshot', () => {
    const snapshot = renderer
      .create(<UniversityView {...props} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('should display the error when error is true', async () => {
    props.error = true;
    const { getByText } = render(
      <UniversityView {...props} />
    );
    const errMsg = await getByText("Whoops! Something went wrong, please try after sometime");
    expect(errMsg).toBeInTheDocument();
  });
});
