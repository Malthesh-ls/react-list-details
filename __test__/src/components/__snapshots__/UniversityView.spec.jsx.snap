// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UniversityView Tests should match snapshot 1`] = `
<div>
  No data found
</div>
`;
